ksql-engine/src/main/java/io/confluent/ksql/internal/KsqlEngineMetrics.java

```diff
@@ -43,11 +46,15 @@
   private final Sensor messageConsumptionByQuery;
   private final Sensor errorRate;
 
+  private final String ksqlServiceId;
+
 
   private final KsqlEngine ksqlEngine;
 
   public KsqlEngineMetrics(String metricGroupPrefix, KsqlEngine ksqlEngine) {
     this.ksqlEngine = ksqlEngine;
+    this.ksqlServiceId = KsqlConstants.KSQL_INTERNAL_TOPIC_PREFIX
+        + ksqlEngine.getKsqlConfig().getString(KsqlConfig.KSQL_SERVICE_ID_CONFIG);
```

Comment:
Should we have some sort of delimiter in this name between the service id and the metric name?

Reply:
We definitely can but I would say if we have such delimiter we need to also have it in the streams/consumer/producer metric names too. Otherwise we will have different naming formats.